plugins/ci/skills/payload-snapshot/scripts/payload_snapshot.py

```
@@ -2543,10 +2561,115 @@ def _collect_rpmdb(
+        """Extract and trim one package's changelog from the target rpmdb."""
+        rel_path = f"{target_tag}/rpmdb/{variant}/changelogs/{pkg}.txt"
+        output_path = os.path.join(variant_dir, "changelogs", f"{pkg}.txt")
+        if os.path.exists(output_path):
+            _log(f"    {variant}/{pkg}: skip (exists)")
+            return {
+                "variant": variant, "package": pkg, "old": old, "new": new,
+                "changelog": rel_path,
+            }
+
+        try:
+            result = subprocess.run(
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)